tests/pr_test.go

```diff
@@ -14,35 +17,127 @@ import (
+	if _, err := os.Stat(outputDir); os.IsNotExist(err) {
+		err := os.Mkdir(outputDir, 0755)
+		if err != nil {
+			return "", err
+		}
+	} else {
+		files, err := os.ReadDir(outputDir)
+		if err != nil {
+			return "", err
+		}
+		for _, f := range files {
+			err = os.Remove(outputDir + "/" + f.Name())
+			if err != nil {
+				return "", err
+			}
+		}
+	}
 
-func TestMain(m *testing.M) {
+	adminStrings := make([]string, 0, len(usernames))
 
-	mydir, err := os.Getwd()
-	if err != nil {
-		fmt.Println(err)
+	for i, username := range usernames {
+		password := uuid.New().String()
+
+		cmd := exec.Command("ibmcloud", "tke", "sigkey-add")
+		cmd.Env = append(os.Environ(), "CLOUDTKEFILES="+outputDir)
+		cmd.Stdin = strings.NewReader(fmt.Sprintf("%s\n%s\n%s\n", username, password, password))
+		cmdOutput, err := cmd.CombinedOutput()
+		if err != nil {
+			return "", fmt.Errorf("error while running 'ibmcloud tke sigkey-add': %v\nCommand output: %s", err, string(cmdOutput))
+		}
+
+		key := fmt.Sprintf("%s/%d.sigkey", outputDir, i+1)
+		adminStrings = append(adminStrings, fmt.Sprintf("{ \"name\": \"%s\", \"key\": \"%s\", \"token\": \"%s\" }", username, key, password))
 	}
-	cmd, err := exec.Command("bash", "./scripts/create_keys.sh", mydir).Output()
+
+	return fmt.Sprintf("[%s]", strings.Join(adminStrings, ", ")), nil
+}
+
+func TestMain(m *testing.M) {
+	cwd, err := os.Getwd()
 	if err != nil {
-		fmt.Printf("error %s", err)
+		log.Fatalf("Error getting current working directory: %v", err)
 	}
-	output := string(cmd)
+	sigDirectory = cwd + "/sigs"
 
-	admin = output
-	// fmt.Print(admin)
 	os.Exit(m.Run())
 }
 
 func TestRunHpcsExample(t *testing.T) {
 	t.Parallel()
 
+	usernames := []string{"admin1"}
+
+	admins, err := CreateSigKeys(usernames, sigDirectory)
+	if !assert.Nilf(t, err, "Error creating sigkeys: %v", err) {
+		log.Fatalf("Error creating sigkeys: %v", err)
+	}
+
 	options := testhelper.TestOptionsDefaultWithVars(&testhelper.TestOptions{
 		Testing:      t,
 		TerraformDir: "examples/complete",
 		Prefix:       "hpcs-crypto",
 		Region:       "us-south",
 		TerraformVars: map[string]interface{}{
-			"admins": "[" + admin + "]",
+			"admins": admins,
+		},
+	})
+
+	output, err := options.RunTestConsistency()
+	assert.Nil(t, err, "This should not have errored")
+	assert.NotNil(t, output, "Expected some output")
+}
+
+func TestRunHpcsFsCloudExample(t *testing.T) {
+	t.Parallel()
+
+	usernames := []string{"admin1", "admin2", "admin3"}
+
+	admins, err := CreateSigKeys(usernames, sigDirectory)
+	if !assert.Nilf(t, err, "Error creating sigkeys: %v", err) {
+		log.Fatalf("Error creating sigkeys: %v", err)
+	}
+
+	options := testhelper.TestOptionsDefaultWithVars(&testhelper.TestOptions{
+		Testing:       t,
+		TerraformDir:  "examples/fscloud",
+		Prefix:        "hpcs",
+		Region:        "us-south",
+		ResourceGroup: "hpcs-dz43ww-resource-group",
+		TerraformVars: map[string]interface{}{
+			"admins": admins,
```

Comment:
Like we did with ICD modules, can we have FSCloud example run for normal test, and Complete example run as upgrade test? Basic example should run in other_test.go. Looks like `hybrid-hpcs` is not tested at all? Perhaps needs to be added to other_test.go?

Reply:
@daniel-butler-irl seems we still need to add other_test.go to test the examples not tested in pr_test.go

Reply:
unrelated to fscloud but I'll add it now since I'm making so many other changes...

Reply:
This will need to be done in a followup PR as more to it than just adding the test. hsm is only available to whitelisted accounts (presumably we have been added for testing) We need an ID (probably private) and I'm not sure what else.